utilities/test_suite/rpp_test_suite_voxel.h

```diff
@@ -550,30 +550,76 @@ void compare_outputs_pkd(Rpp32f* output, Rpp32f* refOutput, int &fileMatch, Rppt
-        outputTempRef = refOutput + Cnt * descriptorPtr3D->strides[0];
-        int height = roiGenericSrcPtr[Cnt].roiHeight;
-        int width = roiGenericSrcPtr[Cnt].roiWidth;
-        int depth = roiGenericSrcPtr[Cnt].roiDepth;
+        outputTemp = output + cnt * descriptorPtr3D->strides[0];
+        outputTempRef = refOutput + cnt * descriptorPtr3D->strides[0];
+        int height = roiGenericSrcPtr[cnt].roiHeight;
+        int width = roiGenericSrcPtr[cnt].roiWidth;
+        int depth = roiGenericSrcPtr[cnt].roiDepth;
         int depthStride = descriptorPtr3D->strides[2];
+        int refDepthStride = descriptorPtr3D->strides[2] * descriptorPtr3D->dims[1];
         int rowStride = descriptorPtr3D->strides[3];
+        int refRowStride = descriptorPtr3D->strides[3] * 3;
         int channelStride = descriptorPtr3D->strides[1];
-        int matched_idx = 0;
+        int matchedIdx = 0;
+
+        for(int c = 0; c < descriptorPtr3D->dims[1]; c++)
+        {
+            outputTempChn = outputTemp + c * channelStride;
+            outputTempRefChn = outputTempRef + c;
+            for(int d = 0; d < depth; d++)
+            {
+                depthTemp = outputTempChn + d * depthStride;
+                depthTempRef = outputTempRefChn + d * refDepthStride;
+                for(int i = 0; i < height; i++)
+                {
+                    rowTemp = depthTemp + i * rowStride;
+                    rowTempRef = depthTempRef + i * refRowStride;
+                    for(int j = 0; j < width; j++)
+                    {
+                        outVal = rowTemp + j;
+                        outRefVal = rowTempRef + j * 3 ;
+                        int diff = abs(*outVal - *outRefVal);
+                        if(diff <= CUTOFF)
+                            matchedIdx++;
+                    }
+                }
+            }
+        }
+        if(matchedIdx >= (1 - TOLERANCE) * (height * width * descriptorPtr3D->dims[1] * depth) && matchedIdx !=0)
+            fileMatch++;
+    }
+}
+
+// compares the output of PLN1-PLN1 variants
+void compare_outputs_pln1(Rpp32f* output, Rpp32f* refOutput, int &fileMatch, RpptGenericDescPtr descriptorPtr3D, RpptRoiXyzwhd *roiGenericSrcPtr)
+{
+    Rpp32f *rowTemp, *rowTempRef, *outVal, *outRefVal, *outputTemp, *outputTempRef, *outputTempChn, *outputTempRefChn, *depthTemp, *depthTempRef;
+    for(int cnt = 0; cnt < descriptorPtr3D->dims[0]; cnt++)
+    {
+        outputTemp = output + cnt * descriptorPtr3D->strides[0];
+        outputTempRef = refOutput + cnt * descriptorPtr3D->strides[0];
+        int height = roiGenericSrcPtr[cnt].roiHeight;
+        int width = roiGenericSrcPtr[cnt].roiWidth;
+        int depth = roiGenericSrcPtr[cnt].roiDepth;
+        int depthStride = descriptorPtr3D->strides[2];
+        int rowStride = descriptorPtr3D->strides[3];
+        int channelStride = descriptorPtr3D->strides[1];
+        int matchedIdx = 0;
 
         for(int c = 0; c < descriptorPtr3D->dims[1]; c++)
```

Comment:
Same comment as before
Need not require dims[1] loop for PLN1

Reply:
Done